src/awkward/_connect/_numba/arrayview.py

```diff
@@ -40,9 +40,9 @@ def find(x):
             for i, array in enumerate(arrays):
                 if x is array:
                     return i
-            else:
```

Comment:
`for` loops do not need an `else` unless there's a `break` in them. It makes the code harder to read and makes the user thing something special is happening. This goes really well with the guard-style (no `else` after `return`/`break`/`raise`), but that's a bigger change for later.

Reply:
I was thinking of the `else` as helpful information for humans reading the code: it's a signal that one can either exit the function in the loop or handle the "no matches" case. (I wouldn't, for instance, use a `for`-`else` patterns for a loop without any nonlocal jumping: `break`, `return`, `raise`.)

It's true that it only makes a semantic difference for `break`, and if that's a reason for not using the `else`, then sure. All of these are weak preferences for me, in which I could have gone either way. If the linter defines a standard, I have no complaints about following it.

Reply:
An else after a loop should tell the reader that there's control flow here. Returns are already control flow. The else is meaningless, it has no impact, but a reader has to read the code above and prove that - plus they also have to try to align the else with the for and not with the nested ifs. Plus they have to realize that the else does nothing. Many Python users don't even know it's valid to use an else with a for; there are special cases (with break) where it's really nice and solves a problem with a teachable structure instead of injecting loop control variables, which is why it's there, but it shouldn't be used if there's no break.

Eventually you'll probably see me argue against else's and elif's that don't have an effect in control flow either, but this one is easier due to the fact it's a rare and specialized feature for `for` and `break`.

Reply:
(If there were a reverse check, one that forced else's after loops that always break, it might be different, because then a reader could take semantic meaning from the else. Currently an always breaking for might not have an else, so every usage has to be read carefully. But it's not what the else was for, and it adds indentation and complexity for the reader, so it's not likely to exist unless you write it yourself. Which is not too bad, from what I'm told.)